src/algebra/curves/alt_bn128/alt_bn128_g1.cpp

```diff
@@ -424,10 +425,10 @@ std::istream& operator>>(std::istream &in, alt_bn128_G1 &g)
 
 #ifdef NO_PT_COMPRESSION
     in >> is_zero >> tX >> tY;
-    is_zero -= '0';
+    is_zero -= static_cast <char>('0');
```

Comment:
Why is the cast needed? The character literal is already a `char` (http://en.cppreference.com/w/cpp/language/character_literal).